src/main/java/uk/gov/ons/ctp/response/collection/exercise/service/impl/BusinessEventValidator.java

```diff
@@ -19,74 +15,99 @@
 
 public class BusinessEventValidator implements EventValidator {
 
-  /**
-   * q Validates the events timestamps are in the correct order and the updated event can be
-   * updated, i.e is not a mandatory or reminder event.
-   *
-   * @param existingEvents
-   * @param updatedEvent
-   * @return
-   */
+  /** Validates the dates on event creation */
   public boolean validate(
       final List<Event> existingEvents,
-      final Event updatedEvent,
+      final Event submittedEvent,
       final CollectionExerciseState collectionExerciseState) {
 
-    // Can only update reminders of the non mandatory events when READY_FOR_LIVE
-    if ((collectionExerciseState.equals(CollectionExerciseState.READY_FOR_LIVE)
-            || collectionExerciseState.equals(CollectionExerciseState.LIVE))
-        && (isMandatory(updatedEvent) || !isReminder(updatedEvent))) {
-      return false;
+    final Map<String, Event> events =
+        existingEvents.stream().collect(Collectors.toMap(Event::getTag, Function.identity()));
+
+    if (isMandatory(submittedEvent)) {
+      return validateMandatoryEvents(events, submittedEvent, collectionExerciseState);
```

Comment:
These looks like "types" of validators that belong in their own classes

Reply:
Done